bundles/org.openhab.binding.fenecon/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,144 @@
+
+	<!-- Sample Thing Type -->
+	<thing-type id="fenecon">
+
+		<label>FENECON Home</label>
+		<description>Thing for FENECON Home Device</description>
+		<category>Solarplant</category>
+
+		<channels>
+			<channel id="state" typeId="state"/>
+			<channel id="lastUpdate" typeId="lastUpdate"/>
+			<channel id="essSoc" typeId="essSoc"/>
+			<channel id="chargerPower" typeId="chargerPower"/>
+			<channel id="dischargerPower" typeId="dischargerPower"/>
+			<channel id="emergencyPowerMode" typeId="emergencyPowerMode"/>
+			<channel id="consumptionActivePower" typeId="consumptionActivePower"/>
+			<channel id="productionActivePower" typeId="productionActivePower"/>
+			<channel id="sellToGridPower" typeId="sellToGridPower"/>
+			<channel id="sellToGridEnergy" typeId="sellToGridEnergy"/>
+			<channel id="buyFromGridPower" typeId="buyFromGridPower"/>
+			<channel id="buyFromGridEnergy" typeId="buyFromGridEnergy"/>
+		</channels>
+
+		<config-description>
+			<parameter name="hostname" type="text" required="true">
+				<context>network-address</context>
+				<label>Hostname</label>
+				<description>Hostname or IP address of the FENECON device, e.g. 192.168.1.11</description>
+			</parameter>
+			<parameter name="password" type="text">
+				<context>password</context>
+				<label>Password</label>
+				<default>user</default>
+				<description>Password to access the device. The password for guest access is set by default.</description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="port" type="integer" min="1">
+				<context>network-address</context>
+				<label>Port</label>
+				<description>Port of the FENECON device</description>
+				<default>8084</default>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="1">
+				<label>Refresh Interval</label>
+				<description>Interval the device is polled in sec.</description>
+				<default>30</default>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Channel Types -->
+	<channel-type id="state">
+		<item-type>String</item-type>
+		<label>System State</label><!-- Use uppercase words, except prepositions. 2-3 words, max 25 chars -->
+		<description>FENECON system state: Ok, Info, Warning or Fault</description>
+		<category>Text</category>
+		<state readOnly="true" pattern="%s"/>
+	</channel-type>
+	<channel-type id="lastUpdate">
+		<item-type>DateTime</item-type>
+		<label>Last Update</label>
+		<description>Last successful update via REST-API from the FENECON system</description>
+		<category>Time</category>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="essSoc">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Battery State</label>
+		<description>Battery state of charge in percent</description>
+		<category>BatteryLevel</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
```

Comment:
```suggestion
		<state readOnly="true" pattern="%.0f %%"/>
```

Reply:
I do not understand this, everywhere in other bindings %unit% is explicitly specified for percent. I have now added a unitHint.

Reply:
Percentage is a little special compared to others. AFAIK the ui hint should also work.

Reply:
%unit% together with the unitHint seems to be the best solution.